blueocean-web/src/main/js/main.jsx

```diff
@@ -217,12 +217,16 @@ function startApp(routes, stores) {
         locationService.setCurrent(newLocation);
     });
 
+    locationService.navCount = 0;
+    window.addEventListener('popstate', e => locationService.navCount-=2); // gets a route change AND popstate
+    const onRouteChange = () => locationService.navCount++;
+
     sseService._initListeners();
 
     // Start React
     render(
         <Provider store={store}>
-            <Router history={history}>{ makeRoutes(routes) }</Router>
+            <Router history={history}><Route onChange={onRouteChange}>{ makeRoutes(routes) }</Route></Router>
```

Comment:
can`t we move `onChange` into the parent?

Reply:
Doesn't seem to support the `onChange` on the `Router` element, only a `Route`